autogen/main/main.tf.tmpl

```diff
@@ -220,13 +220,13 @@ locals {
   # BETA features
   cluster_istio_enabled                    = ! local.cluster_output_istio_disabled
   cluster_dns_cache_enabled                = var.dns_cache
-  cluster_telemetry_type_is_set            = var.cluster_telemetry_type != null
   cluster_pod_security_policy_enabled      = local.cluster_output_pod_security_policy_enabled
   cluster_intranode_visibility_enabled     = local.cluster_output_intranode_visbility_enabled
   confidential_node_config                 = var.enable_confidential_nodes == true ? [{ enabled = true }] : []
 
   # /BETA features
 {% endif %}
+  cluster_telemetry_type_is_set            = var.cluster_telemetry_type != null
```

Comment:
`cluster_telemetry_type` is still a beta feature as per https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#cluster_telemetry